### PR TITLE
refac: organize zk directories

### PR DESCRIPTION
**Pull request by chokobole**
# Description

The following are moved to `zk/plonk/base`

 - `ColumnKey`
- `ColumnType`
- `TableXXX`
- `Phase`

The following are moved to `zk/plonk/constraint_system`

- `Challenge`
- `Circuit`
- `Constraint`
- `ConstraintSystem`
- `Gate`
- `Query`
- `Rotation`
- `SelectorXXX`
- `VirtualCellXXX`

`zk/plonk/circuit` is renamed to `zk/plonk/layout`
example circuits are moved to `zk/plonk/examples`
stringifiers are moved to `zk/plonk/halo2/stringifiers`

### CODE REVIEW

File: tachyon/zk/plonk/layout/floor_planner/v1/v1_strategy.h [hunk 1]
```diff
@@ -66,7 +66,7 @@ SlotInResult<F> SlotIn(std::vector<RegionShape<F>>& region_shapes) {
     // Sort the region's columns to ensure determinism.
     // NOTE(TomTaehoonKim): Sorted result might be different from the original
     // See
-    // https://github.com/kroma-network/halo2/blob/7d0a36990452c8e7ebd600de258420781a9b7917/halo2_proofs/src/circuit/floor_planner/v1/strategy.rs#L171-L191
+    // https://github.com/kroma-network/halo2/blob/7d0a36990452c8e7ebd600de258420781a9b7917/halo2_proofs/src/layout/floor_planner/v1/strategy.rs#L171-L191
```
Comment (dongchangYoo) [inline tachyon/zk/plonk/layout/floor_planner/v1/v1_strategy.h:69]: this link address should not be changed. it is not about the tachyon repo.

**Review by dongchangYoo — CHANGES_REQUESTED** [review]

**Comment by ashjeong** [comment]
Change "Followings are" to "The following are" in the _Description_

**Review by Insun35 — APPROVED** [review]
LGTM

**Comment by TomTaehoonKim** [comment]
@ashjeong FYI, you can just edit PR descriptions by yourself. (Log is left who changed last)

File: tachyon/math/base/BUILD.bazel [hunk 2]
```diff
@@ -141,7 +141,7 @@ tachyon_cc_unittest(
         ":groups",
         ":rational_field",
         ":sign",
-        "//tachyon/base/buffer:buffer",
+        "//tachyon/base/buffer",
```
Comment (TomTaehoonKim) [inline tachyon/math/base/BUILD.bazel:144]: Could you make this change a separate commit?

File: tachyon/zk/plonk/circuit/BUILD.bazel [hunk 3]
```diff
@@ -332,52 +197,24 @@ tachyon_cc_library(
...
+        ":phase",
         ":ref_table",
         ":region_column",
-        ":rotation",
-        ":selector_compressor",
-        ":selector_description",
         "//tachyon/math/elliptic_curves/bn/bn254:g1",
         "//tachyon/math/finite_fields/test:gf7",
-        "//tachyon/math/polynomials/univariate:univariate_polynomial",
+        "//tachyon/math/polynomials/univariate:univariate_evaluations",
```
Comment (TomTaehoonKim) [inline tachyon/zk/plonk/circuit/BUILD.bazel]: Can you make this change as a separate commit?

File: tachyon/zk/plonk/circuit/BUILD.bazel [hunk 4]
```diff
@@ -332,52 +197,24 @@ tachyon_cc_library(
...
-        "rotation_unittest.cc",
-        "selector_compressor_unittest.cc",
-        "selector_description_unittest.cc",
-        "selector_unittest.cc",
     ],
     deps = [
         ":column_key",
-        ":exclusion_matrix",
         ":lookup_table_column",
+        ":phase",
```
Comment (TomTaehoonKim) [inline tachyon/zk/plonk/circuit/BUILD.bazel]:
```suggestion
```

File: vendors/halo2/BUILD.bazel [hunk 5]
```diff
@@ -14,18 +14,18 @@ tachyon_rust_library(
...
         ":bn254_msm",
         ":bn254_msm_gpu",
         ":bn254_poly",
+        ":bn254_rational_evals",
         ":bn254_shplonk_prover",
         ":bn254_shplonk_proving_key",
-        ":bn254_rational_evals",
         ":xor_shift_rng",
         ":xor_shift_rng_cxx_bridge",
+        "//tachyon/rs:tachyon_rs",
```
Comment (TomTaehoonKim) [inline vendors/halo2/BUILD.bazel:28]: These changes will be deleted since #258 is merged right?

**Review by TomTaehoonKim — CHANGES_REQUESTED** [review]

File: vendors/halo2/BUILD.bazel (hunk 5 above)
Comment (chokobole) [reply vendors/halo2/BUILD.bazel]: Yes!

File: tachyon/math/base/BUILD.bazel (hunk 2 above)
Comment (chokobole) [reply tachyon/math/base/BUILD.bazel]: This is resolved automatically since https://github.com/kroma-network/tachyon/pull/258 is merged.

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Comment by ashjeong** [comment]
Can I ask why some commits are "refac(zk)" and some are just "refac", when this is all occurring within "zk"?

**Review by dongchangYoo — APPROVED** [review]
LGTM
